### PR TITLE
Update dcc.Download

### PR DESCRIPTION
**Pull request by tobinngo**
Documents #926

Post-merge checklist:

The master branch is auto-deployed to `dash.plotly.com`.
Once you have merged your PR, wait 5-10 minutes and check dash.plotly.com
to verify that your changes have been made.

- [x] I understand

If this PR documents a new feature of Dash:

- [x] Comment on the original Dash issue with a link to the new docs.

If this PR adds an image or animated GIF:
- [x] This image was saved and referenced locally rather than via an external link

If I introduced a new relative link inside `dcc.Markdown`:
- [x] I considered whether I could replace the `dcc.Markdown` call with `rc.Markdown`, which will replace relative links with `tools.relpath` internally. Otherwise, I used e.g. `<dccLink href=tools.relpath('/layout') children="the first chapter"/>` instead of `[the first chapter](/layout)` (importing `tools` from `dash_docs`), and set `dangerously_allow_html=true` in the `dcc.Markdown` call.

### CODE REVIEW

File: dash_docs/chapters/dash_core_components/Download/index.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    html.H3('Downloading Images'),
+    rc.Syntax(examples['download-image.py'][0], summary=rc.Markdown('''
+        To download a file from disk use `dcc.send_file`, taking care to
+        specify the file path.
+    ''')),
+    rc.Example(examples['download-image.py'][1]),
+
+    html.Hr(),
+
+    html.H3('Downloading Dataframes'),
```
Comment (chriddyp) [inline dash_docs/chapters/dash_core_components/Download/index.py:40]: let's put this one above the images one

File: dash_docs/chapters/dash_core_components/Download/index.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    rc.Syntax(examples['download-dataframe.py'][0], summary=rc.Markdown('''
+        For downloading dataframes the many pandas export methods are supported:
+    ''')),
+    rc.Example(examples['download-dataframe.py'][1]),
+
```
Comment (chriddyp) [inline dash_docs/chapters/dash_core_components/Download/index.py:45]: could we add a download excel file example too, where the excel file is auto-generated from a data frame?

File: dash_docs/chapters/dash_core_components/Download/index.py (hunk 1 above)
Comment (chriddyp) [inline dash_docs/chapters/dash_core_components/Download/index.py:40]: let's also call this download data frame as csv, and then have a separate example for excel file

**Review by chriddyp — APPROVED** [review]
looks good! just one more example for an excel file and we should be good

File: dash_docs/chapters/dash_core_components/Download/index.py (hunk 2 above)
Comment (tobinngo) [reply dash_docs/chapters/dash_core_components/Download/index.py:45]: Should I also add a link to the relevant panda docs?

**Pull request merged**